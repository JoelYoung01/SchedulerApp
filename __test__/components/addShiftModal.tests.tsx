import React, { useEffect } from "react";
import ReactDOM from "react-dom";
import { render, fireEvent } from "@testing-library/react";
