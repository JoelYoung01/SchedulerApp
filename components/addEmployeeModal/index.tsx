export * from "./AddEmployeeModal";
export * from "./AvailiabilityTabs";
export * from "./AvailabilityEditor";
export * from "./EditEmployeeInfo";
export * from "./TabPanel";
