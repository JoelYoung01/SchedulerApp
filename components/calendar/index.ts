export * from "./Calendar";
export * from "./DailyShifts";
export * from "./ShiftCard";
export * from "./ShiftInfo";
export * from "./WeeklyDate";
