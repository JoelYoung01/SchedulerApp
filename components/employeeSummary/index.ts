export * from "./EmployeeSummary";
export * from "./EmployeeCard";
