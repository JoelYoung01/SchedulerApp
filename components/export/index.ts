export * from "./ExportModal";
export * from "./FormatSelect";
