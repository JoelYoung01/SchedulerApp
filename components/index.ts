export * from "./calendar";
export * from "./export";
export * from "./layout";
