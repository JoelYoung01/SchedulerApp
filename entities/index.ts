export * from "./shift";
export * from "./employee";
export * from "./time";
export * from "./types";
export * from "./schedule";
export * from "./color";
