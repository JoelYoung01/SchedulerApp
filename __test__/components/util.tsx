import { expect } from "@jest/globals";
import type { MatcherFunction } from "expect";

import "@testing-library/jest-dom";
import { render, screen, fireEvent, Matcher } from "@testing-library/react";

import React from "react";
import { ExportModal, ExportType } from "../../components";

const toBeHidden: MatcherFunction<[]> = function toBeHidden(actual) {
  if (!(actual instanceof HTMLElement)) {
    throw new Error("actual must be an HTMLElement");
  }
  let current: HTMLElement | null = actual;
  while (current !== null) {
    if (current.style.display === "none") {
      return {
        message: () =>
          "Expected element or parent not to be hidden by `style.display = 'none'`",
        pass: true
      };
    }
    current = current.parentElement;
  }
  return {
    message: () =>
      "Expected element or parent to be hidden by `style.display = 'none'`",
    pass: false
  };
};
expect.extend({ toBeHidden });

declare module "expect" {
  interface AsymmetricMatchers {
    toBeHidden: () => void;
  }
  interface Matchers<R> {
    toBeHidden: () => R;
  }
}

export function checkExportHide(
  name: string,
  element: JSX.Element,
  selector: Matcher
): void {
  test(name, async() => {
    const exportRef: React.RefObject<React.ReactInstance> = React.createRef();
    // Need to add export modal to add printing events
    render(
      <>
        {element}
        <ExportModal
          componentToExport={exportRef}
          defaultValue={ExportType.pdf}
        />
        <div className="printed" data-testid="shown" />
      </>
    );
    expect(screen.getByText(selector)).not.toBeHidden();
    fireEvent(window, new Event("beforeprint"));
    expect(screen.getByText(selector)).toBeHidden();
    expect(screen.getByTestId("shown")).not.toBeHidden();
    fireEvent(window, new Event("afterprint"));
    expect(screen.getByText(selector)).not.toBeHidden();
  });
}
