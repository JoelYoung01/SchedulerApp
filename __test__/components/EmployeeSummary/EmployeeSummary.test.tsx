import { EmployeeSummary } from "../../../components";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import { Color, DayOftheWeek, Employee, Shift, Time } from "../../../entities";

test("Employee Summary Renders", () => {
  const dispatch = jest.fn();
  const setCurrentEmployee = jest.fn();
  const setAddEmployeeModalOpen = jest.fn();
  const employeeSummary = render(
    <EmployeeSummary
      employees={[]}
      dispatch={dispatch}
      currentEmployee={null}
      setCurrentEmployee={setCurrentEmployee}
      addEmployeeModalOpen={false}
      setAddEmployeeModalOpen={setAddEmployeeModalOpen}
    />
  );
  expect(employeeSummary).toMatchSnapshot();
});

test("Employee Summary Renders with employees", () => {
  const dispatch = jest.fn();
  const employee1 = new Employee("Alice", 0, 40, new Color("Pink"), [
    new Shift(
      "",
      new Time(10, DayOftheWeek.Monday),
      new Time(12, DayOftheWeek.Monday)
    )
  ]);
  const employee2 = new Employee("Bob", 0, 40, new Color("Red"), [
    new Shift(
      "",
      new Time(10, DayOftheWeek.Monday),
      new Time(12, DayOftheWeek.Monday)
    )
  ]);
  const setCurrentEmployee = jest.fn();
  const setAddEmployeeModalOpen = jest.fn();
  const employeeSummary = render(
    <EmployeeSummary
      employees={[employee1, employee2]}
      dispatch={dispatch}
      currentEmployee={null}
      setCurrentEmployee={setCurrentEmployee}
      addEmployeeModalOpen={false}
      setAddEmployeeModalOpen={setAddEmployeeModalOpen}
    />
  );
  expect(employeeSummary).toMatchSnapshot();
});

// modal opens and closes with buttons
// modal opens and closes with employee card click
