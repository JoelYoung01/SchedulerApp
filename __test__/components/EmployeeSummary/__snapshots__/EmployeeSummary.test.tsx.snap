// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Employee Summary Renders 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-1bpfpyh"
      >
        <div
          class="css-e53awj-MuiStack-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-ns280z-MuiTypography-root"
          >
            Employees
          </h6>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium css-15n7dva-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            +
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="css-nen11g-MuiStack-root"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-1bpfpyh"
    >
      <div
        class="css-e53awj-MuiStack-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 css-ns280z-MuiTypography-root"
        >
          Employees
        </h6>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium css-15n7dva-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          +
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="css-nen11g-MuiStack-root"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Employee Summary Renders with employees 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-1bpfpyh"
      >
        <div
          class="css-e53awj-MuiStack-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-ns280z-MuiTypography-root"
          >
            Employees
          </h6>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium css-15n7dva-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            +
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="css-nen11g-MuiStack-root"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1mgarw2-MuiPaper-root-MuiCard-root"
          >
            <span
              class="MuiButtonBase-root MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiCardContent-root css-1i89k6q-MuiCardContent-root"
              >
                <div
                  class="css-1q36rmd-MuiStack-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-1rha56l-MuiTypography-root"
                  >
                    Alice
                  </p>
                  <button
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorError MuiIconButton-sizeMedium css-1rbzd5r-MuiButtonBase-root-MuiIconButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="DeleteIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
              <span
                class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
              />
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
          </div>
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-o8cmg4-MuiPaper-root-MuiCard-root"
          >
            <span
              class="MuiButtonBase-root MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiCardContent-root css-1i89k6q-MuiCardContent-root"
              >
                <div
                  class="css-1q36rmd-MuiStack-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-1rha56l-MuiTypography-root"
                  >
                    Bob
                  </p>
                  <button
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorError MuiIconButton-sizeMedium css-1rbzd5r-MuiButtonBase-root-MuiIconButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="DeleteIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
              <span
                class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
              />
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-1bpfpyh"
    >
      <div
        class="css-e53awj-MuiStack-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 css-ns280z-MuiTypography-root"
        >
          Employees
        </h6>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium css-15n7dva-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          +
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="css-nen11g-MuiStack-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1mgarw2-MuiPaper-root-MuiCard-root"
        >
          <span
            class="MuiButtonBase-root MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiCardContent-root css-1i89k6q-MuiCardContent-root"
            >
              <div
                class="css-1q36rmd-MuiStack-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1rha56l-MuiTypography-root"
                >
                  Alice
                </p>
                <button
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorError MuiIconButton-sizeMedium css-1rbzd5r-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="DeleteIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
            <span
              class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
            />
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </span>
        </div>
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-o8cmg4-MuiPaper-root-MuiCard-root"
        >
          <span
            class="MuiButtonBase-root MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiCardContent-root css-1i89k6q-MuiCardContent-root"
            >
              <div
                class="css-1q36rmd-MuiStack-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1rha56l-MuiTypography-root"
                >
                  Bob
                </p>
                <button
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorError MuiIconButton-sizeMedium css-1rbzd5r-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="DeleteIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
            <span
              class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
            />
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </span>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
